Failed to restart cloudcore

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->
**What happened**:
After restarting cloudcore, it throws an error.
```
panic: interface conversion: interface {} is *v1alpha1.ObjectSync, not *v1alpha1.ClusterObjectSync

goroutine 582 [running]:
github.com/kubeedge/kubeedge/pkg/client/listers/reliablesyncs/v1alpha1.(*clusterObjectSyncLister).List.func1({0x189d4c0, 0x40036b0810})
	/go/src/github.com/kubeedge/kubeedge/pkg/client/listers/reliablesyncs/v1alpha1/clusterobjectsync.go:53 +0xf8
github.com/kubeedge/kubeedge/vendor/k8s.io/client-go/tools/cache.ListAll({0x1bcd028, 0x40005ac9d8}, {0x1bc96e0, 0x400159ae10}, 0x4000812d38)
	/go/src/github.com/kubeedge/kubeedge/vendor/k8s.io/client-go/tools/cache/listers.go:40 +0xb8
github.com/kubeedge/kubeedge/pkg/client/listers/reliablesyncs/v1alpha1.(*clusterObjectSyncLister).List(0x40009c1140, {0x1bc96e0, 0x400159ae10})
	/go/src/github.com/kubeedge/kubeedge/pkg/client/listers/reliablesyncs/v1alpha1/clusterobjectsync.go:52 +0x8c
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).deleteClusterObjectSyncs(0x4000805a00)
	/go/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/synccontroller.go:167 +0x64
github.com/kubeedge/kubeedge/cloud/pkg/synccontroller.(*SyncController).Start(0x4000805a00)
	/go/src/github.com/kubeedge/kubeedge/cloud/pkg/synccontroller/synccontroller.go:109 +0x84
github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core.moduleKeeper({0x18e5c3e, 0xe}, 0x4000990990, {{0x18e5c3e, 0xe}, {0x18ddcda, 0x7}, {0x0, {0x0, 0x0}}})
	/go/src/github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core/core.go:77 +0xf0
created by github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core.StartModules
	/go/src/github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core/core.go:45 +0x428
```

**What you expected to happen**:
After restarting cloudcore, it can run correctly.

**How to reproduce it (as minimally and precisely as possible)**:
Restarting cloudcore with clusterobjectsyncs resource present.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.24.13
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`): v1.14.0
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`):
  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>

--- a/pkg/apis/reliablesyncs/v1alpha1/type.go
+++ b/pkg/apis/reliablesyncs/v1alpha1/type.go
@@ -48,7 +48,7 @@ type ClusterObjectSyncList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 
 	// List of ClusterObjectSync.
-	Items []ObjectSync `json:"items"`
+	Items []ClusterObjectSync `json:"items"`
 }
 
 // +genclient

--- a/pkg/apis/reliablesyncs/v1alpha1/zz_generated.deepcopy.go
+++ b/pkg/apis/reliablesyncs/v1alpha1/zz_generated.deepcopy.go
@@ -60,7 +60,7 @@ func (in *ClusterObjectSyncList) DeepCopyInto(out *ClusterObjectSyncList) {
 	in.ListMeta.DeepCopyInto(&out.ListMeta)
 	if in.Items != nil {
 		in, out := &in.Items, &out.Items
-		*out = make([]ObjectSync, len(*in))
+		*out = make([]ClusterObjectSync, len(*in))
 		for i := range *in {
 			(*in)[i].DeepCopyInto(&(*out)[i])
 		}

